### PR TITLE
Dockerized dev environment, geth rinkeby

### PR DESCRIPTION
**Pull request by chapsuk**
Hi, guys! 

1. added optional user/password to db config
1. Added `docker-compose.yml` for running isolated development environment: 
    * postgres
    * migrations
    * geth --rinkeby
    * vulcanizedb

`Vulcanizedb` looking to `geth.ipc` by named volume: `rinkeby_vulcanizedb_geth_data`. Rinkeby long time sync and until full state syncing, the vulcanizedb run failing with message:

```
≻ docker logs -f rinkeby_vulcanizedb
Using config file: /config.toml

2018/06/21 16:54:13 geth initial: state sync not finished
```

Makefile helpful commands:

command | description
--- | ---
rinkeby_env_up | starting geth, postgres and rolling migrations, after migrations done starting vulcanizedb container
rinkeby_env_deploy | build and run vulcanizedb container in dev environment
rinkeby_env_mirate | build and run dev env migrations
rinkeby_env_down | stop and remove all dev env containers

*Note:*
I'm checking migrations and geth, waiting full sync for check vulcanizedb 😴 

```
≻ docker exec -it rinkeby_vulcanizedb_geth geth --rinkeby attach
Welcome to the Geth JavaScript console!

instance: Geth/v1.8.11-stable/linux-amd64/go1.10.3
 modules: admin:1.0 clique:1.0 debug:1.0 eth:1.0 miner:1.0 net:1.0 personal:1.0 rpc:1.0 txpool:1.0 web3:1.0

> eth.syncing
{
  currentBlock: 405692,
  highestBlock: 2502070,
  knownStates: 3505726,
  pulledStates: 3497052,
  startingBlock: 405183
}
```

ref https://github.com/vulcanize/vulcanizedb/issues/16

### CODE REVIEW

**Comment by rmulhol** [comment]
@chapsuk looks solid, thanks! Could you add some instructions to the README about these commands?

Will be curious to see how it looks after you get geth synced. I suspect the timeframe for doing so will vary based on one's machine, but would be interested to hear roughly how long we'd expect it to take to get the docker env setup

**Comment by chapsuk** [comment]
@rmulhol syncing done,  it took about 10-16h  😄 
I think without geth cpu limit  It can take much less time.

After full sync and start valucanizedb all looks good:

```
≻ docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS                    PORTS                                 NAMES
10bdf9d1499f        rinkeby_vulcanizedb          "/app sync --startin…"   6 minutes ago       Up 6 minutes                                                    rinkeby_vulcanizedb
b17ca2fd4ecb        postgres:9.6.5-alpine        "docker-entrypoint.s…"   14 minutes ago      Up 14 minutes (healthy)   5432/tcp                              rinkeby_vulcanizedb_postgres
7262d9e5d372        ethereum/client-go:v1.8.11   "geth --rinkeby --rp…"   14 minutes ago      Up 14 minutes             8545-8546/tcp, 30303/tcp, 30303/udp   rinkeby_vulcanizedb_geth

≻ docker logs -f rinkeby_vulcanizedb
Using config file: /config.toml

2018/06/22 15:22:51 Backfilling 2507392 blocks

Validating Blocks
|2507377|-- Validation Window --|2507392| (2507392:HEAD)

Validating Blocks
|2507378|-- Validation Window --|2507393| (2507393:HEAD)

Validating Blocks
|2507378|-- Validation Window --|2507393| (2507393:HEAD)

Validating Blocks
|2507379|-- Validation Window --|2507394| (2507394:HEAD)

Validating Blocks
|2507379|-- Validation Window --|2507394| (2507394:HEAD)

Validating Blocks
|2507380|-- Validation Window --|2507395| (2507395:HEAD)

Validating Blocks
|2507380|-- Validation Window --|2507395| (2507395:HEAD)
...
^C

≻ docker exec -it rinkeby_vulcanizedb_postgres psql -U postgres -d vulcanizedb
psql (9.6.5)
Type "help" for help.

vulcanizedb=# select count(*) from blocks;
 count
--------
 158204
(1 row)

vulcanizedb=# select count(*) from transactions;
 count
-------
  2404
(1 row)

```

additionally added optional user/password to db config
PTAL

**Comment by chapsuk** [comment]
@rmulhol can you please review?

works good for me, vdb running 3 days without any issue, current sync state:

```
vulcanizedb=# select min(number), max(number), count(*) from blocks;
 min |   max   |  count
-----+---------+---------
   0 | 2525445 | 2146351
(1 row)
```

```
≻ docker ps | grep vulcanizedb
10bdf9d1499f        rinkeby_vulcanizedb          "/app sync --startin…"   3 days ago          Up 3 days                                                                                                rinkeby_vulcanizedb
b17ca2fd4ecb        postgres:9.6.5-alpine        "docker-entrypoint.s…"   3 days ago          Up 3 days (healthy)   5432/tcp                                                                           rinkeby_vulcanizedb_postgres
7262d9e5d372        ethereum/client-go:v1.8.11   "geth --rinkeby --rp…"   3 days ago          Up 3 days             8545-8546/tcp, 30303/tcp, 30303/udp                                                rinkeby_vulcanizedb_geth

≻ docker inspect rinkeby_vulcanizedb_geth -f '{{.RestartCount}}'
0

≻ docker inspect rinkeby_vulcanizedb -f '{{.RestartCount}}'
0

≻ docker inspect rinkeby_vulcanizedb_postgres -f '{{.RestartCount}}'
0
```

File: dockerfiles/rinkeby/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    depends_on:
+      postgres:
+        condition: service_healthy
+    command: -database postgresql://postgres:postgres@postgres:5432/vulcanizedb?sslmode=disable -path /migrations up
+    volumes:
+      - ./../../db/migrations:/migrations
+    networks:
+      vulcanizedb_net:
+
+  postgres:
```
Comment (gustin) [inline dockerfiles/rinkeby/docker-compose.yml]: thanks for the PR @chapsuk! Would a volume or volume option make sense for the postgres container so the data is retained if the container is stopped?

Comment (gustin) [reply dockerfiles/rinkeby/docker-compose.yml]: Awesome, thanks for the clarification.

**Pull request merged**